rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -150,4 +168,58 @@ public Project convertToEmbeddedWithExternalIds(Project sw360Object) {
         TProtocol protocol = new TCompactProtocol(thriftClient);
         return new ProjectService.Client(protocol);
     }
+
+    public Function<ProjectLink, ProjectLink> filterAndSortAttachments(Collection<AttachmentType> attachmentTypes) {
+        Predicate<Attachment> filter = att -> attachmentTypes.contains(att.getAttachmentType());
+        return createProjectLinkMapper(rl -> rl.setAttachments(nullToEmptyList(rl.getAttachments())
+                .stream()
+                .filter(filter)
+                .sorted(Comparator
+                        .comparing((Attachment a) -> nullToEmpty(a.getCreatedTeam()))
+                        .thenComparing(Comparator.comparing((Attachment a) -> nullToEmpty(a.getCreatedOn())).reversed()))
+                .collect(Collectors.toList())));
+    }
+
+    public Function<ProjectLink, ProjectLink> createProjectLinkMapper(Function<ReleaseLink, ReleaseLink> releaseLinkMapper){
+        return (projectLink) -> {
+            List<ReleaseLink> mappedReleaseLinks = nullToEmptyList(projectLink
+                    .getLinkedReleases())
+                    .stream()
+                    .map(releaseLinkMapper)
+                    .collect(Collectors.toList());
+            projectLink.setLinkedReleases(mappedReleaseLinks);
+            return projectLink;
+        };
+    }
+
+    protected List<ProjectLink> createLinkedProjects(Project project, Function<ProjectLink, ProjectLink> projectLinkMapper, boolean deep,
+            User user) {
+        final Collection<ProjectLink> linkedProjects = flattenProjectLinkTree(getLinkedProjects(project, deep, user));
```

Comment:
Can we re-use `flattenProjectLinkTree` and `getLinkedProjects` method present in `SW360Utils`?

Reply:
Done